src/page.js

```diff
@@ -14,10 +14,12 @@ class UnrequestedPage {
   get isSettled() { return false; }
 
   get records() {
-    var records = this.data.map(function (content, index) {
-      return new Record(this, content, index);
-    }, this);
-    return records;
+    if (!this._records) {
```

Comment:
very clever I like it!
I already merged my own PR for this, but let's try and merge in this change along with it!